lib/fs.js

```diff
@@ -552,10 +551,11 @@ fs.readFileSync = function(path, options) {
   var isUserFd = isFd(path); // file descriptor ownership
   var fd = isUserFd ? path : fs.openSync(path, options.flag || 'r', 0o666);
 
+  tryStatSync(fd, isUserFd);
```

Comment:
Ack, since it's an patch level enhancement that should land in `v8.x`, although it will probably be relevant for 1 minor `node8` release.